test/src/specs/sync/ibd_process.rs

```diff
@@ -25,21 +25,58 @@ impl Spec for IBDProcess {
         node0.connect(node2);
         node0.connect(node3);
         node0.connect(node4);
+        // The node's outbound connection does not retain the peer which in the ibd state
+        node0.generate_blocks(1);
         // will never connect
         node0.connect_uncheck(node5);
         node0.connect_uncheck(node6);
-        node0.generate_blocks(1);
 
         sleep(Duration::from_secs(5));
 
-        let rpc_client = node0.rpc_client();
+        let rpc_client0 = node0.rpc_client();
         let ret = wait_until(10, || {
-            let peers = rpc_client.get_peers();
+            let peers = rpc_client0.get_peers();
             peers.len() == 4
         });
 
         if !ret {
             panic!("refuse to connect fail");
         }
+
+        // IBD only with outbound/whitelist node
+        let rpc_client1 = node1.rpc_client();
+        let rpc_client2 = node2.rpc_client();
+        let rpc_client3 = node3.rpc_client();
+        let rpc_client4 = node4.rpc_client();
+        let rpc_client5 = node5.rpc_client();
+        let rpc_client6 = node6.rpc_client();
+
+        let ret = wait_until(10, || {
+            let header1 = rpc_client1.get_tip_header();
+            let header2 = rpc_client2.get_tip_header();
+            let header3 = rpc_client3.get_tip_header();
+            let header4 = rpc_client4.get_tip_header();
+            let header5 = rpc_client5.get_tip_header();
+            let header6 = rpc_client6.get_tip_header();
+            !(header1.inner.number.0 != 0
```

Comment:
Double negation is hard to read.

Reply:
I think so too, but clippy tells me this way faster, I will change it back and disable clippy warning